src/components/FullscreenModal/FullscreenModal.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { isUndefined, noop } from 'ramda-adjunct';
+
+import { getColor, getDepth } from '../../utils/helpers';
+import ModalHeader from './Header/Header';
+import ModalFooter from './Footer/Footer';
+import { FullscreenModalLayouts } from './FullscreenModal.enums';
+import { ColumnConfigMap, FullscreenModalProps } from './FullscreenModal.types';
+import { Col, Container, Row } from '../layout';
+
+const BaseModal = styled.div`
+  width: 100vw;
+  height: 100vh;
+  position: fixed;
+  top: 0;
+  left: 0;
+  overflow-y: auto;
+  background-color: ${getColor('graphite5H')};
+  z-index: ${getDepth('modal')};
+`;
+
+const columnConfigMap: ColumnConfigMap = {
+  [FullscreenModalLayouts.single6]: {
+    header: [6, 3],
+    sidebar: [0, 0],
+    content: [6, 3],
+  },
+  [FullscreenModalLayouts.single8]: {
+    header: [8, 2],
+    sidebar: [0, 0],
+    content: [8, 2],
+  },
+  [FullscreenModalLayouts.sidebar46]: {
+    header: [10, 1],
+    sidebar: [4, 1],
+    content: [6, 0],
+  },
+  [FullscreenModalLayouts.sidebar48]: {
+    header: [12, 0],
+    sidebar: [4, 0],
+    content: [8, 0],
+  },
+};
+
+const FullscreenModal: React.FC<FullscreenModalProps> = ({
+  layout = FullscreenModalLayouts.single6,
+  header,
+  content,
+  sidebar,
+  footer,
+  onClose = noop,
+}) => {
+  const modalRef = useRef<HTMLDivElement>(null);
+
+  const {
+    header: headerConfig,
+    sidebar: sidebarConfig,
+    content: contentConfig,
+  } = columnConfigMap[layout];
+
+  const hasLayoutSidebar = sidebarConfig[0] > 0;
+  const totalContentWidth = contentConfig[0] + sidebarConfig[0];
+
+  if (hasLayoutSidebar && isUndefined(sidebar)) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      `You chose to use modal layout with sidebar (current: ${layout}) but you didn't provide sidebar content.
+You should either provide content in "sidebar" property or switch layout to "${FullscreenModalLayouts.single6}" or "${FullscreenModalLayouts.single8}"
+`,
+    );
```

Comment:
could we throw an Error here? (if this isn't expected usage, I would rather not allow it)

Reply:
So I was thinking about this and I think that throwing an error is too strong since this isn't a violation which could break the application, I personally would leave throwing for things which could cause actual errors in the application.
However, I did change `console.warn` to `console.error`.

Reply:
If the modal is passed `layout={.."sidebar"}` but the comp is not supplied in the `sidebar` prop -> that should throw Error in my opinion as well as it sounds like a user mistake. Console logs/errors get lost easily

Reply:
Ok guys change to throw error :)